src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1331,9 +1331,20 @@ bool isRefEscape
             // widest possible escape via writeable ref-like receiver or ref/out argument.
             uint escapeTo = scopeOfTheContainingExpression;
 
+            var isReadOnlyInvocation = symbol switch
+            {
+                MethodSymbol m => m.IsEffectivelyReadOnly,
+                // TODO: val escape checks should be skipped for property accesses when
```

Comment:
This is the case you were telling me about, where we can't easily determine which accessor we are using?

So an indexer that takes a `Span<T>` (for multi-dimensional indexing) won't work with this change and will need to be handled in the future, correct (just talking about the getter, since the setter needs to be mutable in this case)?

Reply:
Yeah if it looks like `this[Span<T> span] { readonly get; set; }` we will be pessimistic and say the span could escape illegally even though we should be able to tell that only the getter is called.